### DISCUSSION TITLE
How to use Razzle with Cypress?

### DISCUSSION THREAD

**Discussion by silviubogan** [jaredpalmer/razzle#1700] [Q&A] [answered]
Hello!

I have this in my package.json, inside the scripts section:

```json
"cy:open": "npx concurrently \"CYPRESS=true npm start\" \"npx cypress open\""
```

My problem is: the process `npm start` does not let the `npx cypress open` process to know when it is prepared to receive requests, so I have to wait about 15 seconds after the `npx cypress open` command opens its window before I click the `Try again` button in it, and only then I am able to click on a test suite and let it run.

![image](https://user-images.githubusercontent.com/198924/130634485-cc28288a-4b3b-4047-b312-02dcc0c1a229.png)

Can you tell me a better way to do it?

Thanks.

**Comment by fivethreeo** [comment]
https://github.com/jaredpalmer/razzle/discussions/1633

**Reply by silviubogan** [reply]
Great answer, but my question is also about E2E tests (integration tests) in Cypress, not only about unit tests.

**Comment by SimeonC** [comment, marked as answer]
Hey, we do cypress and razzle with the following package scripts. Hopefully they can help you out.

```
"cypress:open": "cypress open",
"start:test": "PORT=8089 NODE_CONFIG_ENV=test razzle start",
"test:integration:dev": "start-server-and-test start:test http://localhost:8089/ cypress:open",
```

**Reply by fivethreeo** [reply]
https://www.npmjs.com/package/start-server-and-test
